lib/fluent/plugin/in_http.rb

```diff
@@ -355,46 +318,30 @@ def on_message_complete
 
         if @keep_alive
           header['Connection'] = 'Keep-Alive'
-          send_response(code, header, body)
-        else
-          send_response_and_close(code, header, body)
         end
-      end
-
-      def on_write_complete
-        close if @next_close
-      end
-
-      def send_response_and_close(code, header, body)
         send_response(code, header, body)
-        @next_close = true
       end
 
-      def closing?
-        @next_close
-      end
+      def send_response(code, header, body=nil)
+        unless @keep_alive
+          @io_handler.closing = true # keepalive disabled for HTTP/1.0 (or earlier)
+        end
 
-      def send_response(code, header, body)
-        header['Content-length'] ||= body.bytesize
-        header['Content-type'] ||= 'text/plain'
+        if body
+          header['Content-length'] ||= body.bytesize
+          header['Content-type'] ||= 'text/plain'
```

Comment:
In RFC 7231, content type header is described as [Content-**T**ype](http://tools.ietf.org/html/rfc7231#section-3.1.1.5).